services/UserSocialMediaService.ts

```diff
@@ -0,0 +1,58 @@
+import { Service } from 'typedi';
+import { EntityManager } from 'typeorm';
+import { InjectManager } from 'typeorm-typedi-extensions';
+import { ForbiddenError, NotFoundError } from 'routing-controllers';
+import { UserError } from '../utils/Errors';
+import { UserSocialMediaModel } from '../models/UserSocialMediaModel';
+import { UserModel } from '../models/UserModel';
+import Repositories, { TransactionsManager } from '../repositories';
+import { Uuid, PublicUserSocialMedia, SocialMedia } from '../types';
+
+@Service()
+export default class UserSocialMediaService {
+  private transactions: TransactionsManager;
+
+  constructor(@InjectManager() entityManager: EntityManager) {
+    this.transactions = new TransactionsManager(entityManager);
+  }
+
+  public async getSocialMediaForUser(user: UserModel): Promise<PublicUserSocialMedia[]> {
+    const userSocialMedia = await this.transactions.readOnly(async (txn) => Repositories.userSocialMedia(txn)
+      .getSocialMediaForUser(user));
+    return userSocialMedia.map((sm) => sm.getPublicSocialMedia());
+  }
+
+  public async insertSocialMediaForUser(user: UserModel, socialMedia: SocialMedia) {
+    const addedSocialMedia = await this.transactions.readWrite(async (txn) => {
+      const userSocialMediaRepository = Repositories.userSocialMedia(txn);
+      const isNewSocialMediaType = await userSocialMediaRepository.isNewSocialMediaTypeForUser(user, socialMedia.type);
+      if (!isNewSocialMediaType) throw new UserError('Social media URL of this type has been created for this user');
```

Comment:
nit: Let's add an "already" here so that it explains why this case is an error (since otherwise, the error message here makes it sound like the social media entry has been created, even though an error is thrown)
```suggestion
      if (!isNewSocialMediaType) throw new UserError('Social media URL of this type has already been created for this user');
```